### DISCUSSION TITLE
Analyzer disabling discussion continued

### DISCUSSION THREAD

**Discussion by robertson-bauercontrols** [EWSoftware/VSSpellChecker#298] [General]
Hi, sorry to be a thorn in your side, thought I could start a discussion here instead of in the issues.  I'm attempting to disable the C# code analyzer, so I can have it spellcheck C# code the same way it does comments for now.  Based on the feedback, I disabled the analysis checkbox in general settings. 

However, at that point, the code analysis was disabled, except it also wasn't spellchecking the code (the comments where still getting checked).  There's something here I'm missing.

```
// solution editor.config
vsspell_section_id = 8a725214157e41da8d6edd5e4472eaff
vsspell_spell_check_as_you_type = true
vsspell_include_in_project_spell_check = true
vsspell_code_analyzers_enabled = false
vsspell_detect_doubled_words = true
vsspell_ignore_words_with_digits = false
vsspell_ignore_words_in_all_uppercase = false
vsspell_ignore_words_in_mixed_case = false
vsspell_ignore_format_specifiers = true
vsspell_ignore_filenames_and_email_addresses = true
vsspell_ignore_xml_elements_in_text = true
vsspell_treat_underscore_as_separator = true
vsspell_ignore_mnemonics = false
vsspell_ignored_character_class = NonLatin
vsspell_code_analyzer_ignore_xml_doc_comments = false
vsspell_code_analyzer_ignore_delimited_comments = false
vsspell_code_analyzer_ignore_standard_single_line_comments = false
vsspell_code_analyzer_ignore_quadruple_slash_comments = false
vsspell_code_analyzer_ignore_normal_strings = false
vsspell_code_analyzer_ignore_verbatim_strings = false
vsspell_code_analyzer_ignore_interpolated_strings = false
vsspell_code_analyzer_apply_to_all_c_style_languages = true
vsspell_determine_resource_file_language_from_name = false
vsspell_additional_dictionary_folders_8a725214157e41da8d6edd5e4472eaff = clear_inherited|.\.spellcheck\custom_dictionaries
vsspell_dictionary_languages_8a725214157e41da8d6edd5e4472eaff = en-US
vsspell_ignored_words_8a725214157e41da8d6edd5e4472eaff = clear_inherited|File:.\.spellcheck\solution_ignored_words.dic
```

**Comment by EWSoftware** [comment]
The code analyzer only spell checks the identifiers in the code.  It doesn't spell check anything else.  Comments and strings are still handled by the tagger.  I did not move spell checking of those elements over to the analyzer due to its limitations with adding words to the dictionaries and ignored words files.  By disabling the code analyzer, it effectively works as it has done in the past.

**Comment by robertson-bauercontrols** [comment]
I think I explained it poorly. I'm wondering if I can get the code to behave the same as comments.

I'd like to do the "spell check solution" menupic, and have that dialog show both comments and code. No analyzers at all 🙂

I appreciate the response btw. This spell checker blows away any other one I've tried, even paid ones. It's quality.
________________________________

**Comment by EWSoftware** [comment]
Solution/project spell check is implemented purely in code that I wrote and doesn't involve any of the Visual Studio APIs.  It uses a series of regular expressions to extract things that look like strings and comments from the files in the project [classifying them by extension](https://ewsoftware.github.io/VSSpellChecker/html/0ff35371-69b5-48dd-a062-037abe2469de.htm).  Implementing support for identifiers based on the language would be too difficult since you would have to parse the actual code in some way based on the language.  I'm not sure of a good way to accomplish that.

**Reply by robertson-bauercontrols** [reply]
Could I classify all .cs files as PlainTextClassifier? Then it would just check everything in the file.

**Comment by EWSoftware** [comment]
You could but I think you'd have way too many false positives to have it be of any use since it won't apply any of the C# options that control how it spell checks the comments and strings under normal circumstances.  There are instructions in the referenced topic for replacing a handler with one of your own in the user configuration file so you could give it a try.  You'd just add an entry to remap the "cs" extension to the "PlainText" classifier.

**Comment by glen-nicol** [comment]
Sorry if this is off topic but it seems related. I've been trying to figure out how to configure ```VSSpell001``` and ```VSSpell002``` to be a suggestion instead of a warning via my .editorconfig file. However, I cannot find the complete magic string to get that to work. Any specific help? The lightbulb menu for configuring it never produces a result (spins forever and some errors show up in visual studio's top bar). And I've looked through the documentation for the analyzer settings but don't see any specifics on how to configure the level of the two rules just the various cases where it shows up.

I can create an issue for the bad configuration behavior but I may not be able to produce any logs for it.

**Comment by EWSoftware** [comment]
```
[*.{cs,vb}]
# Treat spell checker issues as suggestions rather than warnings (or errors if treating all warnings as errors)
dotnet_diagnostic.VSSpell001.severity = suggestion

# Or turn off the code analyzer completely so that no identifier spell checking is performed
vsspell_code_analyzers_enabled = false
```

You can also set it to silent or none which gets rid of all the markers but that's effectively turning it off so you might as well use the spelling configuration option.

**Reply by glen-nicol** [reply]
That did it! I had tried that before with no results. Normally configuring those has immediate results in the editor. This time I tried building after the change and the squiggles went away and the warnings moved to the suggestion category in the error list as desired. 

Thanks!
